legacy/firmware/solana.c

```
@@ -169,11 +183,143 @@
+      break;
+    default:
+      fsm_sendFailure(FailureType_Failure_DataError,
+                      "Message format must be 0 or 1");
+      return false;
+  }
+  return true;
+}
+//=====================Offchain Message Signing=====================
+
+//=====================Unsafe Message Signing=====================
+bool solana_sign_unsafe_message(const SolanaSignUnsafeMessage *msg,
```

Cppcheck (2.10-2)

[style] The function 'solana_sign_unsafe_message' is never used.

(unusedFunction)